internal/command/machine/select.go

```diff
@@ -0,0 +1,53 @@
+package machine
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/superfly/flyctl/api"
+	"github.com/superfly/flyctl/flaps"
+	"github.com/superfly/flyctl/internal/appconfig"
+	"github.com/superfly/flyctl/internal/flag"
+)
+
+func checkSelectArgs(ctx context.Context, haveMachineIDs bool) error {
+	selectFlag := flag.GetBool(ctx, "select")
+	appName := appconfig.NameFromContext(ctx)
+	switch {
+	case haveMachineIDs && selectFlag:
+		return errors.New("machine IDs can't be used with --select")
+	case !haveMachineIDs && !selectFlag:
+		return errors.New("a machine ID must be provided unless --select is used")
+	case selectFlag && appName == "":
+		return errors.New("an app name must be specified to use --select")
+	default:
+		return nil
+	}
+}
+
+func selectMachine(ctx context.Context) (*api.Machine, error) {
+	flapsClient := flaps.FromContext(ctx)
+	machines, err := flapsClient.List(ctx, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get a list of machines: %w", err)
+	}
+
+	var machineOptions []string
+	for _, machine := range machines {
+		machineOptions = append(machineOptions, fmt.Sprintf("%s %s", machine.ID, machine.Name))
+	}
+
+	prompt := &survey.Select{
+		Message:  "Select machine:",
+		Options:  machineOptions,
+		PageSize: 15,
+	}
+	var selected int
+	if err := survey.AskOne(prompt, &selected); err != nil {
```

Comment:
let's use our prompt package instead of a 3p package. we have handled a bunch of terminal/platform things in our package.

I think prompt.Select() is the thing to use here.